File: src/main/java/macro303/keschet/pieces/Spearman.java
package macro303.keschet.pieces;

import macro303.keschet.Colour;
import macro303.keschet.Direction;
import org.jetbrains.annotations.NotNull;

import static macro303.keschet.Direction.*;

/**
 * Created by Macro303 on 2018-02-12.
 */
public class Spearman extends Piece {
	@NotNull
//	private static final String symbol = "\u2694";
	private static final String symbol = "P";
	private static final int maxDistance = 2;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, EAST, SOUTH, WEST};

	public Spearman(@NotNull Colour teamColour) {
		super(teamColour, maxDistance, symbol, validDirections);
	}

	public Spearman() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Spearman{}";
	}
}

File: src/main/java/macro303/keschet/pieces/Spearman.kt
package macro303.keschet.pieces

import macro303.keschet.Console
import macro303.keschet.Direction

class Spearman(teamColour: Console.Colour) : Piece(teamColour = teamColour, maxDistance = 2, symbol = "P", validDirections = arrayOf(Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST)) {

	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Spearman) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode(): Int {
		return super.hashCode()
	}

	override fun toString(): String {
		return "Spearman() ${super.toString()}"
	}
}